lib/portlayer/storage/vsphere/image.go

```diff
@@ -366,3 +456,41 @@ func (v *ImageStore) ListImages(ctx context.Context, store *url.URL, IDs []strin
 
 	return images, nil
 }
+
+// find any inprogress.vmdk files, and delete their directories.
+func (v *ImageStore) cleanup(ctx context.Context, store *url.URL) error {
+
+	storeName, err := util.ImageStoreName(store)
+	if err != nil {
+		return err
+	}
+
+	inprogress := &types.HostDatastoreBrowserSearchSpec{
+		MatchPattern: []string{"inprogress.vmdk"},
+	}
+
+	res, err := v.ds.LsDirs(ctx, v.imageStorePath(storeName), inprogress)
+	if err != nil {
+		return err
+	}
+
+	fm := object.NewFileManager(v.s.Vim25())
+
+	for _, fdir := range res.HostDatastoreBrowserSearchResults {
+		if len(fdir.File) == 0 {
+			break
+		}
+
+		log.Infof("Removing in progress image %s", fdir.FolderPath)
+
+		err = tasks.Wait(context.TODO(), func(ctx context.Context) (tasks.Waiter, error) {
+			return fm.DeleteDatastoreFile(ctx, fdir.FolderPath, v.s.Datacenter)
```

Comment:
Possibly learning experience for me. Should you try to remove all the in progress files before reporting any errors trying to remove any of them? Then only the in progress images that you had problems removing would be left.